### DISCUSSION TITLE
How to configure a network between 2 dockers containers belonging to 2 different dockers context ?

### DISCUSSION THREAD

**Discussion by gduvalsc** [abiosoft/colima#941] [General]
Hello, I'm using colima on a Mac OS/X M1. I have a docker image "ora23c" which is located in a context using the X86_64 architecture.

I have another docker container which is using the aarch architecture.

$ colima ls
PROFILE    STATUS     ARCH       CPUS    MEMORY    DISK     RUNTIME    ADDRESS
default    Running    x86_64     8       16GiB     60GiB    docker     
arm        Running    aarch64    2       2GiB      60GiB    docker    

The docker container running in "arm" profile is the client, and I want to connect this client to the oracle database running in the "default" profile on the same host.

So I have 2 differents "docker" environnements and I tried to create a network between them.

No solution using a "bridge" network  unless I miss something... 

I tried too to create an overlay network with "docker swarm init" in one docker and "docker swarm join" in the other docker context. No sucess...

Is there a solution ? Do i miss something ?

Regards.

Gerard

**Comment by gduvalsc** [comment]
I found a solution but I was constrained to rebuild the "colima" environements.

When creating a colima container, the best advice to give to any user is the specify a "network address".

Exemple:

colima start -c 8 -m 16 -a x86_64 --network-address x86_64
colima start -c 8 -m 16 --network-address aarch64

With this example, colima ls will return the following:

PROFILE  STATUS   ARCH    CPUS  MEMORY  DISK   RUNTIME  ADDRESS
aarch64  Running  aarch64  8    16GiB   60GiB  docker   192.168.106.6
x86_64   Running  x86_64   8    16GiB   60GiB  docker   192.168.106.5

The important thing is the address 192.168.106.xxx

Without --network-address, there no ip adress and the containers cannot communicate through an IP address

After when you create the "swarn" with docker swarn init, you need to specify the address of the colima container where you init the swarm.

**Reply by abiosoft** [reply]
What version of Colima are you on? If it is v0.6.x you can check the ip address of each instance with `ip addr show eth0`.
That ip address is reachable from other docker contexts.

**Reply by gduvalsc** [reply]
Sorry for the delay to answer, but my colima version is 5.5

Now with the solution I described above, everything is OK. I can now, from a container located in colima x85_64, address a container located in the other colima using the container name.

I wouldn't have had this problem if the --network-address was the default option.

Regards

**Comment by NicoMincuzzi** [comment]
Hi, I have the same issue. I followed what you described above, creating two different contexts. However it's not clear how I can use a single network for both contexts. It seems that the two containers in the two different contexts not communicate between them. Thanks in advance !

**Comment by gduvalsc** [comment]
Hello,

Once you have created your 2 colima environements, you need to create a swarm (docker swarm init in one colima context).

With the command "docker swarm join-token manager", you will have the command to run in the other context.

You need to have 2 managers in each context. One is created with "docker swarm init" and the other with "docker swarm join".

When you run docker swarm init, you have an IP address to specify. Use "colima ls", to know which address to specify.

After this, you can create a "network" in one or the other context (docker network create ...). This network will be automatically seen in the other context.

Regards

**Reply by NicoMincuzzi** [reply]
Hi,
Thanks for your answer. At the moment, I'm using two docker-compose files, one for each context. In the first one, I'm using a docker container which is using the aarch architecture, while the second one contains a docker image, which is located in a context using the X86_64 architecture.
I tried to create in one of docker-compose file a network with driver bridge in order to linked the two contexts, but the containers seem as not communicate between them. 

Thanks in advance!

**Comment by gduvalsc** [comment]
You must use « overlay » for your network and not « bridge » RegardsDe :  Nicola Mincuzzi ***@***.***>Date :  lundi, 1 janvier 2024 à 10:59À :  abiosoft/colima ***@***.***>Cc :  gduvalsc ***@***.***>, Author ***@***.***>Objet :  Re: [abiosoft/colima] How to configure a network between 2 dockers containers belonging to 2 different dockers context ? (Discussion #941)
Hi,
Thanks for your answer. At the moment, I'm using two docker-compose files, one for each context. In the first one, I'm using a docker container which is using the aarch architecture, while the second one contains a docker image, which is located in a context using the X86_64 architecture.
I tried to create in one of docker-compose file a network with driver bridge in order to linked the two contexts, but the containers seem as not communicate between them.
Thanks in advance!

**Reply by NicoMincuzzi** [reply]
In this particular case, I don't use Swarm and consequently I can't use overlay

**Comment by gduvalsc** [comment]
I’m afraid that there is no solution if you can’t use Swarm!RegardsGerardDe :  Nicola Mincuzzi ***@***.***>Date :  mercredi, 10 janvier 2024 à 19:05À :  abiosoft/colima ***@***.***>Cc :  gduvalsc ***@***.***>, Author ***@***.***>Objet :  Re: [abiosoft/colima] How to configure a network between 2 dockers containers belonging to 2 different dockers context ? (Discussion #941)
In this particular case, I don't use Swarm and consequently I can't use overlay
